### PR TITLE
gui: Replace hardcoded LLE module list with present dumped modules

### PR DESCRIPTION
**Pull request by Zangetsu38**
special thx to elad for help me with some things.

### CODE REVIEW

File: vita3k/gui/src/settings_dialog.cpp [hunk 1]
```diff
@@ -101,7 +71,23 @@ static bool clear_and_refresh_game_list(HostState &host, GuiState &gui) {
...
-using namespace list;
+void get_modules_list(GuiState &gui, HostState &host)
+{
+    if (!gui.modules_list.empty())
+        gui.modules_list.clear();
+
+    static const auto mudules_path = fs::path (host.pref_path) / "vs0/sys/external/";
+    if (!mudules_path.empty()) {
+        fs::recursive_directory_iterator end;
+        static const std::string ext = ".suprx";
```
Comment (VelocityRa) [inline vita3k/gui/src/settings_dialog.cpp]:
```suggestion
        const std::string ext = ".suprx";
```

File: vita3k/gui/src/settings_dialog.cpp [hunk 2]
```diff
@@ -101,7 +71,23 @@ static bool clear_and_refresh_game_list(HostState &host, GuiState &gui) {
     return true;
 }
 
-using namespace list;
+void get_modules_list(GuiState &gui, HostState &host)
+{
+    if (!gui.modules_list.empty())
+        gui.modules_list.clear();
```
Comment (VelocityRa) [inline vita3k/gui/src/settings_dialog.cpp]:
```suggestion
    gui.modules_list.clear();
```

File: vita3k/gui/src/settings_dialog.cpp [hunk 3]
```diff
@@ -101,7 +71,23 @@ static bool clear_and_refresh_game_list(HostState &host, GuiState &gui) {
...
+
+    static const auto mudules_path = fs::path (host.pref_path) / "vs0/sys/external/";
```
Comment (VelocityRa) [inline vita3k/gui/src/settings_dialog.cpp]:
```suggestion
    const auto modules_path = fs::path(host.pref_path) / "vs0/sys/external/";
```

File: vita3k/gui/src/settings_dialog.cpp [hunk 4]
```diff
@@ -101,7 +71,23 @@ static bool clear_and_refresh_game_list(HostState &host, GuiState &gui) {
...
+    if (!mudules_path.empty()) {
+        fs::recursive_directory_iterator end;
+        static const std::string ext = ".suprx";
+        for (fs::recursive_directory_iterator m(mudules_path); m != end; ++m) {
+            const fs::path lle = (*m);
```
Comment (VelocityRa) [inline vita3k/gui/src/settings_dialog.cpp]:
```suggestion
            const fs::path lle = *m;
```

File: vita3k/gui/src/settings_dialog.cpp [hunk 5]
```diff
@@ -112,21 +98,23 @@ void draw_settings_dialog(GuiState &gui, HostState &host) {
...
+        if (!gui.modules_list.empty()) {
+            ImGui::TextColored(GUI_COLOR_TEXT, "Module List");
             if (ImGui::IsItemHovered())
                 ImGui::SetTooltip("Select your desired modules.");
             ImGui::PushItemWidth(240);
-            static bool module_select[MODULES_COUNT] = { false };
-            if (ImGui::ListBoxHeader("##modules_list", MODULES_COUNT, 6)) {
-                for (int m = 0; m < MODULES_COUNT; m++) {
-                    auto modules = std::find(host.cfg.lle_modules.begin(), host.cfg.lle_modules.end(), LIST_MODULES[m]);
+            static char *modules_list = new char[gui.modules_list.size()];
```
Comment (VelocityRa) [inline vita3k/gui/src/settings_dialog.cpp]: Why manual memory management? This isn't C.
This should be a `std::string`.
```suggestion
            static std::string modules_list;
```

File: vita3k/gui/src/settings_dialog.cpp [hunk 6]
```diff
@@ -112,21 +98,23 @@ void draw_settings_dialog(GuiState &gui, HostState &host) {
...
+            static bool* module_select = new bool[gui.modules_list.size()]{ false };
```
Comment (VelocityRa) [inline vita3k/gui/src/settings_dialog.cpp]: Ditto.
```suggestion
            static std::vector<bool> module_select;
```

File: vita3k/gui/src/settings_dialog.cpp [hunk 7]
```diff
@@ -112,21 +98,23 @@ void draw_settings_dialog(GuiState &gui, HostState &host) {
...
+            if (ImGui::ListBoxHeader("##modules_list", static_cast<int>(gui.modules_list.size()), 6)) {
+                for (int m = 0; m < gui.modules_list.size(); m++) {
+                    memcpy(modules_list, gui.modules_list[m].c_str(), gui.modules_list.size());
```
Comment (VelocityRa) [inline vita3k/gui/src/settings_dialog.cpp]:
```suggestion
                    modules_list = gui.modules_list[m];
```

File: vita3k/gui/src/settings_dialog.cpp [hunk 8]
```diff
@@ -112,21 +98,23 @@ void draw_settings_dialog(GuiState &gui, HostState &host) {
...
+                    auto modules = std::find(host.cfg.lle_modules.begin(), host.cfg.lle_modules.end(), gui.modules_list[m]);
                     if (modules != host.cfg.lle_modules.end() && !module_select[m])
                         module_select[m] = true;
-                    if (!gui.module_search_bar.PassFilter(LIST_MODULES[m]))
+                    if (!gui.module_search_bar.PassFilter(modules_list))
```
Comment (VelocityRa) [inline vita3k/gui/src/settings_dialog.cpp]:
```suggestion
                    if (!gui.module_search_bar.PassFilter(modules_list.c_str()))
```

File: vita3k/gui/src/settings_dialog.cpp [hunk 9]
```diff
@@ -112,21 +98,23 @@ void draw_settings_dialog(GuiState &gui, HostState &host) {
...
                         continue;
-                    if (ImGui::Selectable(LIST_MODULES[m], &module_select[m]) && modules == host.cfg.lle_modules.end())
-                        host.cfg.lle_modules.push_back(LIST_MODULES[m]);
+                    if (ImGui::Selectable(modules_list, &module_select[m]) && modules == host.cfg.lle_modules.end())
```
Comment (VelocityRa) [inline vita3k/gui/src/settings_dialog.cpp]:
```suggestion
                    if (ImGui::Selectable(modules_list.c_str(), &module_select[m]) && modules == host.cfg.lle_modules.end())
```

File: vita3k/gui/src/settings_dialog.cpp (hunk 5 above)
Comment (Zangetsu38) [reply vita3k/gui/src/settings_dialog.cpp]: Is elad have send me this code for convert vector string inside char*

Comment (Zangetsu38) [reply vita3k/gui/src/settings_dialog.cpp]: Not forget ImGui::selectable isnonly compatible with char*

File: vita3k/gui/src/settings_dialog.cpp (hunk 6 above)
Comment (Zangetsu38) [reply vita3k/gui/src/settings_dialog.cpp]: Same elad have send me this for bool can using  new char*

File: vita3k/gui/src/settings_dialog.cpp (hunk 5 above)
Comment (VelocityRa) [reply vita3k/gui/src/settings_dialog.cpp]: ...No I did not forget. You can still do it with `.c_str()` I even mentioned it.

Comment (1whatleytay) [reply vita3k/gui/src/settings_dialog.cpp]: They made a whole new operator in c++ and noone uses it :cry:

Comment (Zangetsu38) [reply vita3k/gui/src/settings_dialog.cpp]: I see if you said i check that after back on my computer

**Comment by Zangetsu38** [comment]
@VelocityRa you can suggest name of commit ? I have write it but not sur is very good lol

**Comment by VelocityRa** [comment]
@Zangetsu38 `gui: Replace hardcoded LLE module list with present dumped modules`

**Comment by Zangetsu38** [comment]
Oki thx

File: vita3k/gui/src/settings_dialog.cpp (hunk 3 above)
Comment (Zangetsu38) [reply vita3k/gui/src/settings_dialog.cpp]: Mudules 😂 why i have not see that lol

**Pull request merged**